tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Conventions/PipelineConvention.cs

```diff
@@ -270,6 +283,29 @@ protected virtual Task<bool> ApplyConventionAsync(BuildDefinition definition, Sd
                 hasChanges = true;
             }
 
+
+            if (IsScheduled)
+            {
+                var scheduleTriggers = definition.Triggers.OfType<ScheduleTrigger>();
+                var computedSchedule = CreateScheduleFromDefinition(definition);
+
+                // Here we are basically say that if you don't have any triggers, or the triggers are empty or if it doesn't
+                // match what we computed, then recreate it. This will force consistency but only require an update IF it
+                // doesn't match what we want it to be.
+                if (scheduleTriggers == default
+                        || !scheduleTriggers.Any()
+                        || (scheduleTriggers.First().Schedules[0].StartHours != computedSchedule.StartHours
+                            || scheduleTriggers.First().Schedules[0].StartMinutes != computedSchedule.StartMinutes))
+                {
+                    definition.Triggers.Add(new ScheduleTrigger
```

Comment:
This looks like the errors we were hitting. You always seem to be adding a new schedule trigger instead of just mutating the existing one if there is one.